"Free" price not properly labeled

https://travis-ci.org/facundoolano/google-play-scraper/builds/372416218

``` 
 1) App method should fetch valid application data:
     AssertionError: expected '' to equal 'Free'
      at gplay.app.then (test/lib.app.js:43:16)
      at process._tickCallback (internal/process/next_tick.js:178:7)
  2) List method should fetch apps with fullDetail:
     AssertionError: expected '' to equal 'Free'
      at apps.map (test/lib.list.js:93:14)
      at Array.map (<anonymous>)
      at gplay.list.then.then (test/lib.list.js:75:26)
      at process._tickCallback (internal/process/next_tick.js:178:7)
```

--- a/.gitignore
+++ b/.gitignore
@@ -29,3 +29,4 @@ build/Release
 # Dependency directory
 # https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
 node_modules
+.nyc_output

--- a/lib/app.js
+++ b/lib/app.js
@@ -98,7 +98,10 @@ const MAPPINGS = {
     fun: (val) => val === 0
   },
   currency: ['ds:8', 0, 2, 0, 0, 0, 1, 0, 1],
-  priceText: ['ds:8', 0, 2, 0, 0, 0, 1, 0, 2],
+  priceText: {
+    path: ['ds:8', 0, 2, 0, 0, 0, 1, 0, 2],
+    fun: priceTextForFree
+  },
   offersIAP: {
     path: ['ds:3', 0, 12, 12, 0],
     fun: Boolean
@@ -174,6 +177,14 @@ function descriptionText (description) {
   // preserve the line breaks when converting to text
   const html = cheerio.load('<div>' + description.replace(/<br>/g, '\r\n') + '</div>');
   return cheerio.text(html('div'));
+}
+
+function priceTextForFree (priceText) {
+  // Return Free if the price text is empty
+  if (!priceText) {
+    return 'Free';
+  }
+  return priceText;
 }
 
 function cleanInt (number) {

--- a/package.json
+++ b/package.json
@@ -4,7 +4,8 @@
   "description": "scrapes app data from google play store",
   "main": "index.js",
   "scripts": {
-    "test": "mocha --timeout 5000 && eslint ."
+    "test": "mocha --timeout 5000 && eslint .",
+    "coverage": "nyc mocha --timeout 5000 && eslint ."
   },
   "repository": {
     "type": "git",
@@ -37,6 +38,7 @@
     "eslint-plugin-promise": "^1.1.0",
     "eslint-plugin-standard": "^1.3.2",
     "mocha": "^2.4.5",
+    "nyc": "^11.7.1",
     "promise-log": "^0.1.0",
     "validator": "^5.2.0"
   }

--- a/test/lib.app.js
+++ b/test/lib.app.js
@@ -112,4 +112,21 @@ describe('App method', () => {
      .catch((err) => {
        assert.equal(err.message, 'App not found (404)');
      }));
+
+  it('should reject the promise when appId is not passed', () =>
+    gplay.app({Testkey: 'com.dxco.pandavszombiesasdadad'})
+     .then(() => {
+       throw Error('should not resolve');
+     })
+     .catch((err) => {
+       assert.equal(err.message, 'appId missing');
+     }));
+
+  it('should fetch PriceText for paid apps properly', () => {
+    return gplay.app({appId: 'com.teslacoilsw.launcher.prime', country: 'in'})
+     .then((app) => {
+       assert.equal(app.priceText, '₹ 99.00');
+       assert.equal(app.currency, 'INR');
+     });
+  });
 });

